File: src/main/C#/Consumer.cs
using System;
using Apache.NMS;
using Apache.NMS.Util;
using Apache.NMS.ActiveMQ;

namespace Apache.NMS.ActiveMQ.Book.Ch8
{
   public class Consumer
   {
      public static void Main(string[] args)
      {
	 NMSConnectionFactory NMSFactory = new NMSConnectionFactory("tcp://localhost:61616");
	 IConnection connection = NMSFactory.CreateConnection();
	 ISession session = connection.CreateSession(AcknowledgementMode.AutoAcknowledge);
	 IDestination destination = session.GetTopic("STOCKS.JAVA");
	 IMessageConsumer consumer = session.CreateConsumer(destination);
	 consumer.Listener += new MessageListener(OnMessage);
	 connection.Start();
         Console.WriteLine("Press any key to quit.");
         Console.ReadKey();
      }
   
      protected static void OnMessage(IMessage message)
      {
	ITextMessage TextMessage = message as ITextMessage;
	Console.WriteLine(TextMessage.Text);
      }
   }
}

File: examples/src/main/java/org/apache/activemq/book/ch3/jobs/Consumer.java
package org.apache.activemq.book.ch3.jobs;

import javax.jms.Connection;
import javax.jms.ConnectionFactory;
import javax.jms.Destination;
import javax.jms.JMSException;
import javax.jms.MessageConsumer;
import javax.jms.Session;

import org.apache.activemq.ActiveMQConnectionFactory;

public class Consumer {

    private static String brokerURL = "tcp://localhost:61616";
    private static transient ConnectionFactory factory;
    private transient Connection connection;
    private transient Session session;
    
    private String jobs[] = new String[]{"suspend", "delete"};
    
    public Consumer() throws JMSException {
    	factory = new ActiveMQConnectionFactory(brokerURL);
    	connection = factory.createConnection();
        connection.start();
        session = connection.createSession(false, Session.AUTO_ACKNOWLEDGE);
    }
    
    public void close() throws JMSException {
        if (connection != null) {
            connection.close();
        }
    }    
    
    public static void main(String[] args) throws JMSException {
    	Consumer consumer = new Consumer();
    	for (String job : consumer.jobs) {
    		Destination destination = consumer.getSession().createQueue("JOBS." + job);
    		MessageConsumer messageConsumer = consumer.getSession().createConsumer(destination);
    		messageConsumer.setMessageListener(new Listener(job));
    	}
    }
	
	public Session getSession() {
		return session;
	}


}
